### PR TITLE
Lite: Floor-Div different datatypes supported

### PR DESCRIPTION
**Pull request by ANSHUMAN87**
TFLite FloorDiv currently supports only Int32.
This PR add support for other datatypes as well.
Test cases also updated.

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @aselle: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 29 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/lite/kernels/floor_div.cc [hunk 1]
```diff
@@ -123,8 +130,21 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
...
     }
+    case kTfLiteFloat32: {
+      return EvalImpl<float>(context, data->requires_broadcast, input1, input2,
+                             output);
+    }
+    case kTfLiteUInt8: {
+      return EvalImpl<uint8_t>(context, data->requires_broadcast, input1,
+                               input2, output);
+    }
+    case kTfLiteInt64: {
```
Comment (jdduke) [inline tensorflow/lite/kernels/floor_div.cc]: In general we only add new types when they're explicitly needed. Unless you have an int64 model, let's leave that type out.

File: tensorflow/lite/kernels/floor_div.cc [hunk 2]
```diff
@@ -123,8 +130,21 @@ TfLiteStatus Eval(TfLiteContext* context, TfLiteNode* node) {
       return EvalImpl<int32_t>(context, data->requires_broadcast, input1,
                                input2, output);
     }
+    case kTfLiteFloat32: {
+      return EvalImpl<float>(context, data->requires_broadcast, input1, input2,
+                             output);
+    }
+    case kTfLiteUInt8: {
```
Comment (jdduke) [inline tensorflow/lite/kernels/floor_div.cc]: For uint8, it also requires changing behavior in the toco toolchain to allow the type to be quantized.

**Review by jdduke — CHANGES_REQUESTED** [review]

**Review by jdduke — CHANGES_REQUESTED** [review]
We'll need to update the op versioning as done in https://github.com/tensorflow/tensorflow/commit/e476315d4484ed2b344983e1c5258b976927257c#diff-c6d779ac1662f9870bf80c4a641ca49b. Do you have a particular model which requires this change?

File: tensorflow/lite/kernels/floor_div.cc [hunk 3]
```diff
@@ -64,9 +64,15 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
   TF_LITE_ENSURE_EQ(context, input1->type, input2->type);
 
   const TfLiteType type = input1->type;
-  if (type != kTfLiteInt32) {
-    context->ReportError(context, "Currently floor_div only supports int32.");
-    return kTfLiteError;
+  switch (type) {
+    case kTfLiteFloat32:
+    case kTfLiteInt32:
+    case kTfLiteUInt8:
```
Comment (jdduke) [inline tensorflow/lite/kernels/floor_div.cc]: Can we leave out uint8 and just add float?

**Comment by ANSHUMAN87** [comment]
@jdduke : Your comments are addressed now, Thanks!

**Review by jdduke — APPROVED** [review]

**Comment by ANSHUMAN87** [comment]
@rthadur : Gentle Reminder!

**Pull request merged**